main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala

```diff
@@ -1480,6 +1480,119 @@ object ParsedAst {
 
   }
 
+  /**
+    * Effect Set
+    */
+  sealed trait EffectSet
+
+  object EffectSet {
+    /**
+      * Singleton effect set.
+      *
+      * @param sp1 the position of the first character in the set.
+      * @param eff the effect.
+      * @param sp2 the position of the last character in the set.
+      */
+    case class Singleton(sp1: SourcePosition, eff: Effect, sp2: SourcePosition) extends EffectSet
+
+    /**
+      * Pure effect set.
+      *
+      * @param sp1 the position of the first character in the set.
+      * @param sp2 the position of the last character in the set.
+      */
+    case class Pure(sp1: SourcePosition, sp2: SourcePosition) extends EffectSet
+
+    /**
+      * A set of effects.
+      *
+      * @param sp1  the position of the first character in the set.
+      * @param effs the effects.
+      * @param sp2  the position of the last character in the set.
+      */
+    case class Set(sp1: SourcePosition, effs: Seq[Effect], sp2: SourcePosition) extends EffectSet
+  }
+
+  /**
+    * A single effect.
+    */
+  sealed trait Effect
+
+  object Effect {
+    /**
+      * Effect variable.
+      *
+      * @param sp1   the position of the first character in the effect.
+      * @param ident the name of the variable.
+      * @param sp2   the position of the last character in the effect.
+      */
+    case class Var(sp1: SourcePosition, ident: Name.Ident, sp2: SourcePosition) extends ParsedAst.Effect
+
+    /**
```

Comment:
How about these names and in this order:

- Complement
- Union
- Intersection
- Difference